test/libponyc/badpony.cc

```diff
@@ -1087,6 +1087,41 @@ TEST_F(BadPonyTest, IsComparingCreateSugar)
   }
 }
 
+TEST_F(BadPonyTest, IsComparingCreate)
+{
+  // From issue #4162, this is just the desugared
+  // version of the previous test.
+  const char* src =
```

Comment:
to go along with the missed cases that Joe mentions in his other comment, we need to also test with a constructor that isn't new.

So something like

```pony
class D
  new foo() => None
```

then as part of a test, that...

```pony
D.foo() is D.foo()
```

gives our error

For the sake of completeness, we should also the same with an `actor` and also an `actor` with a custom named constructor like `foo`.

Reply:
Yes! The existence of constructors which aren't called `new` was even documented here: https://tutorial.ponylang.io/types/classes.html#constructors

I'm still a newbie when it comes to Pony. Ok, let me work on it.